language/diem-framework/modules/DiemVMConfig.move

```diff
@@ -218,6 +218,18 @@ module DiemFramework::DiemVMConfig {
 
     /// # Access Control
 
+    /// The permission "UpdateVMConfig" is granted to DiemRoot [[H11]][PERMISSION].
+    spec module {
+        invariant [global, isolated] forall addr: address where exists<DiemConfig<DiemVMConfig>>(addr):
+            addr == @DiemRoot;
+
+        invariant update old(DiemConfig::spec_is_published<DiemVMConfig>())
```

Comment:
I noticed that some specs condition the conclusion whether a resource is published both in old state and current states (as in here), and some just check the old state.   Is this a historical accident, or do we really need to condition on exists in both old and current state sometimes?

Reply:
That's a good point! This is related to the issue of partial functions in spec. To be on a safe side, I think we should check the existence of a certain resource on both the old state and the current state. In some cases (like this case), it's ok to check it on the old state only when there is a "resource persistence" invariant in place, so the existence on the current state is implied.